lib/src/core/credentials/ace_credentials.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 The NAMIB Project Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import 'dart:typed_data';
+
+import '../../definitions/security/ace_security_scheme.dart';
+import 'credentials.dart';
+
+/// [Credentials] used for the [AceSecurityScheme].
+abstract class AceCredentials extends Credentials {
+  /// Constructor.
+  AceCredentials() : super('ace:ACESecurityScheme');
+}
+
+/// Credentials for the ACE-OAuth DTLS profile using a Pre-Shared Key (PSK).
+///
+/// The Credentials need to be obtained from an authorization server.
+class DtlsPskAceCredentials extends AceCredentials {
+  /// Constructor.
+  DtlsPskAceCredentials(this.identity, this.preSharedKey) : super();
+
+  /// The identity used for establishing the DTLS connection.
+  final Uint8List identity;
+
+  /// The pre-shared key used for establishing the DTLS connection.
+  final Uint8List preSharedKey;
+}
```

Comment:
In order to simplify the implementation, I first got the impression that using credentials classes for each profile/credentials type instead of accepting the token class could make sense. This has the downside of placing a bit more burden on the library user, though.

Another solution I am thinking about right now would be accepting the access token in its serialized form instead. This would probably be more flexible.

Reply:
Maybe a combination would make sense here: I suggest we add the flexibility you mentioned by making `AceCredentials` non-abstract and including the serialized access token as a field on it, while still keeping the subclasses like `DtlsPskAceCredentials` for better ease of use. That way, the user can decide whether to work on the serialized access token directly or to use our convenience credential subclasses (though the documentation of `AceCredentials` should make it clear these subclasses exist, otherwise users of this library might miss them if the return type is simply `AceCredentials`).

Reply:
Thanks again for the feedback :) I now followed the serialized approach and I generally have a good feeling about this. I was a bit unsure how to deal with the `DtlsPskAceCredentials` in this scenario, though, as we would need to make the `accessToken` field nullable in this case (I suppose), therefore I commented it out for now. I'll revisit it later :)